apps/settings/src/components/Users/NewUserModal.vue

```diff
@@ -0,0 +1,461 @@
+					:options="subAdminsGroups"
+					v-model="newUser.subAdminsGroups"
+					:close-on-select="false"
+					:multiple="true"
+					label="name" />
+			</div>
+			<div class="modal__item">
+				<label class="modal__label"
+					for="new-user-quota">
+					{{ t('settings', 'Quota') }}
+				</label>
+				<NcSelect class="modal__select"
+					input-id="new-user-quota"
+					:placeholder="t('settings', 'Set user quota')"
+					:options="quotaOptions"
+					v-model="newUser.quota"
+					:clearable="false"
+					:taggable="true"
+					:create-option="validateQuota" />
+			</div>
+			<div v-if="showConfig.showLanguages"
+				class="modal__item">
+				<label class="modal__label"
+					for="new-user-language">
+					{{ t('settings', 'Language') }}
+				</label>
+				<NcSelect	class="modal__select"
+					input-id="new-user-language"
+					:placeholder="t('settings', 'Set default language')"
+					:clearable="false"
+					:selectable="option => !option.languages"
+					:filter-by="languageFilterBy"
+					:options="languages"
+					v-model="newUser.language"
+					label="name" />
+			</div>
+			<div :class="['modal__item managers', { 'icon-loading-small': loading.manager }]">
+				<label class="modal__label"
+					for="new-user-manager">
+					{{ t('settings', 'Manager') }}
+				</label>
+				<NcSelect class="modal__select"
+					input-id="new-user-manager"
+					:placeholder="t('settings', 'Set user manager')"
+					:options="possibleManagers"
+					v-model="newUser.manager"
+					:user-select="true"
+					label="displayname"
+					@search="searchUserManager" />
+			</div>
+			<NcButton class="modal__submit"
+				data-test="submit"
+				type="primary"
+				native-type="submit">
+				{{ t('settings', 'Add new user') }}
+			</NcButton>
+		</form>
+	</NcModal>
+</template>
+
+<script>
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcPasswordField from '@nextcloud/vue/dist/Components/NcPasswordField.js'
+import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+const unlimitedQuota = {
+	id: 'none',
+	label: t('settings', 'Unlimited'),
+}
+
+const defaultQuota = {
+	id: 'default',
+	label: t('settings', 'Default quota'),
+}
+
+export default {
+	name: 'NewUserModal',
+
+	components: {
+		NcButton,
+		NcModal,
+		NcPasswordField,
+		NcSelect,
+		NcTextField,
+	},
+
+	props: {
+		loading: {
+			type: Object,
+			required: true,
+		},
+
+		newUser: {
+			type: Object,
+			required: true,
+		},
+
+		showConfig: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			possibleManagers: [],
+		}
+	},
+
+	computed: {
+		settings() {
+			return this.$store.getters.getServerData
+		},
+
+		usernameLabel() {
+			if (this.settings.newUserGenerateUserID) {
+				return t('settings', 'Username will be autogenerated')
+			}
+			return t('settings', 'Username (required)')
```

Comment:
Small suggestion (not sure if we use that somewhere but would possibly make sense to make clear that it cannot be changed afterwards?):

```suggestion
			return t('settings', 'User id (required)')
```

Reply:
@nimishavijay?